### DISCUSSION TITLE
스테이지와 태스크의 차이를 잘 모르겠어요.

### DISCUSSION THREAD

**Discussion by madplay** [NMP-Study/spark2023#25] [Q&A] [answered]
스테이지와 태스크의 차이를 잘 모르겠습니다. 이론/실무적으로 어떠한 차이가 있을까요?

**Comment by qlqhqo2341** [comment, marked as answer]
제가 경험적으로 얻은 결론을 말씀드리면..
스테이지는 `셔플(분산된 데이터 간의 데이터 교환, 넓은 트랜스포메이션 같은)`이 필요하지 않은 정도의 연산을 모아둔 것
태스크는 스테이지에서, 데이터 파티셔닝까지 쪼개져서 완전히 단일 머신에서 실행 가능한 단위
요렇게 보고있습니다~
